_posts/dotnet-development-on-apple-silicon.md

```diff
@@ -0,0 +1,75 @@
+
+I have two Dell 32" 4K screens (U3219Q) claiming to support DisplayPort over the USB-C connection. But things are not so easy. I'm not sure why, but it sounds like external screens and projectors are more or less always a source of pain with Apple machines.
+
+The problem manifests itself at boot time. The screen is in standby mode, the machine starts, the screen wakes up and starts the USB-C handshaking process, but that happens too early, and the response from the Mac times out. The screen never retries and never connects. The "easy" workaround is disconnecting and reconnecting the USB-C cable, and the screen immediately starts working. However, if you're using the screen as a USB hub, that causes a mess with USB devices connected to the screen. Dell forums are full of people complaining about this issue, so it's not something new to the MacStudio, and it's not easy to understand who to blame. Anyway, the definitive solution is to use a USB-C to DisplayPort cable. No more problems. The downside is that to use the screens as a USB hub, I had to connect them using the USB-A cable, which is fine because the MacStudio comes with two USB-A ports.
+
+Enough talking about the hardware.
+
+## .NET development
+
+_2022-04-15 - One week into using macOS and an M1 for .NET Development._
+
+Let's start with the easy stuff. [JetBrains Rider](https://www.jetbrains.com/rider/) is a joy. .NET 6, runtime and SDK, [are supported on AMR64](https://dotnet.microsoft.com/en-us/download/dotnet/6.0). We can build and execute all .NET projects targeting .NET, including .NET Core. Obviously, except for WFP and Windows Forms projects. They require Windows anyway.
+
+We can install .NET Core 3.1. There is support only for the [x64 version](https://dotnet.microsoft.com/en-us/download/dotnet/3.1). Not a big deal since it goes out of support soon.
+
+Last but not least, [VS Code](https://code.visualstudio.com) works great. Its support for [dockerized environments](https://milestone.topics.it/2021/06/11/visual-studio-code-remote-containers.html) comes in handy.
+
+### Windows virtualization using Parallels
+
+For everything else, .NET Framework related, we need a virtual machine or some way to use Windows. Forget about Bootcamp. It doesn't exist, and it probably won't ever exist for ARM.
+
+I installed the trial version of [Parallels](https://www.parallels.com/it/products/desktop/trial/). One caveat for using virtual machines on ARM hosts is that the virtualized operating system needs to be for ARM64. That might open to issues in virtualizing Windows because there is an [agreement between Microsoft and Qualcomm](https://www.theverge.com/2021/11/23/22798231/microsoft-qualcomm-exclusivity-deal-windows-on-arm) that might affect our ability to license Windows for AMR on non-Qualcomm hardware.

```

Comment:
```suggestion
I installed the trial version of [Parallels](https://www.parallels.com/it/products/desktop/trial/). One caveat for using virtual machines on ARM hosts is that the virtualized operating system needs to be for ARM64. That might open to issues in virtualizing Windows because there is an [agreement between Microsoft and Qualcomm](https://www.theverge.com/2021/11/23/22798231/microsoft-qualcomm-exclusivity-deal-windows-on-arm) that might affect our ability to license Windows for ARM on non-Qualcomm hardware.

```